src/Kernel32.Shared/Kernel32+OVERLAPPED.cs

```diff
@@ -0,0 +1,74 @@
+
+namespace PInvoke
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Contains the <see cref="OVERLAPPED"/> nested class.
+    /// </content>
+    public partial class Kernel32
+    {
+        /// <summary>Contains information used in asynchronous (or overlapped) input and output (I/O).</summary>
+        /// <remarks>
+        /// Any unused members of this structure should always be initialized to zero before the structure is used in a function
+        /// call. Otherwise, the function may fail and return <see cref="Win32ErrorCode.ERROR_INVALID_PARAMETER"/>.
+        /// <para>
+        /// The Offset and OffsetHigh members together represent a 64-bit file position.It is a byte offset from the start of
+        /// the file or file-like device, and it is specified by the user; the system will not modify these values.The calling
+        /// process must set this member before passing the OVERLAPPED structure to functions that use an offset, such as the
+        /// ReadFile or WriteFile (and related) functions.
+        /// </para>
+        /// <para>
+        /// You can use the HasOverlappedIoCompleted macro to check whether an asynchronous I/O operation has completed if
+        /// GetOverlappedResult is too cumbersome for your application.
+        /// </para>
+        /// <para>You can use the CancelIo function to cancel an asynchronous I/O operation.</para>
+        /// <para>
+        /// A common mistake is to reuse an OVERLAPPED structure before the previous asynchronous operation has been
+        /// completed. You should use a separate structure for each request. You should also create an event object for each thread
+        /// that processes data. If you store the event handles in an array, you could easily wait for all events to be signaled
+        /// using the WaitForMultipleObjects function.
+        /// </para>
+        /// </remarks>
+        [StructLayout(LayoutKind.Sequential)]
+        public struct OVERLAPPED
+        {
+            /// <summary>
+            /// The status code for the I/O request. When the request is issued, the system sets this member to STATUS_PENDING to indicate that the operation has not yet started. When the request is completed, the system sets this member to the status code for the completed request.
+            /// <para>The Internal member was originally reserved for system use and its behavior may change.</para>
+            /// </summary>
+            public IntPtr Internal;
+
+            /// <summary>
+            /// The number of bytes transferred for the I/O request. The system sets this member if the request is completed without errors.
+            /// <para>The InternalHigh member was originally reserved for system use and its behavior may change.</para>
+            /// </summary>
+            public IntPtr InternalHigh;
```

Comment:
It's defined as `ULONG_PTR` in the native object so it must be 64-bits on 64-bits archs and 32-bits on 32-bits ones. If we fix it to one size it'll corrupt memory on one of the architectures.

It could map to what Mono call a `native uint` but the non-Mono C# only support `IntPtr` as a native-pointer-sized type.

The CoreClr version is here : [Overlapped.cs](https://github.com/dotnet/coreclr/blob/bc146608854d1db9cdbcc0b08029a87754e12b49/src/mscorlib/src/System/Threading/Overlapped.cs#L46) (We could use it if we ignore Windows 8 as [MSDN](https://msdn.microsoft.com/en-us/library/system.threading.nativeoverlapped(v=vs.110).aspx) seem to say that it's available to store apps only since WIn 10)